### PR TITLE
Add RelayEnvironment in public API

### LINKED ISSUE #1025: Add RelayEnvironment to the public API
Now that #558 has landed, and as pointed out in #898, it would be very useful to let people create new environments on their own such as:
```
const environment = new Relay.Environment();
````

### PR DESCRIPTION
**Pull request by tlvenn**
This resolves #1025

### CODE REVIEW

**Comment by tlvenn** [comment]
CLA signed.

**Comment by josephsavona** [comment]
Thanks for submitting this. We're on board with adding this to the public API, but can you also export these other modules as well? `RelayEnvironment` can't be used w/o one of the new "root" components:
- `Relay.Environment` (`RelayEnvironment`)
- `Relay.Renderer` (`RelayRenderer`)
- `Relay.ReadyStateRenderer` (`RelayReadyStateRenderer`)

**Comment by tlvenn** [comment]
Done

File: src/RelayPublic.js [hunk 1]
```diff
@@ -21,6 +21,9 @@ const RelayRootContainer = require('RelayRootContainer');
 const RelayRoute = require('RelayRoute');
 const RelayStore = require('RelayStore');
 const RelayInternals = require('RelayInternals');
+const RelayEnvironment = require('RelayEnvironment');
+const RelayRenderer = require('RelayRenderer');
+const RelayReadyStateRenderer = require('RelayReadyStateRenderer');
```
Comment (josephsavona) [inline src/RelayPublic.js]: nit: alpha ordering, please! ;-)

File: src/RelayPublic.js [hunk 2]
```diff
@@ -41,6 +44,9 @@ const RelayPublic = {
   RootContainer: RelayRootContainer,
   Route: RelayRoute,
   Store: RelayStore,
+  Environment: RelayEnvironment,
+  Renderer: RelayRenderer,
+  ReadyStateRenderer: RelayReadyStateRenderer,
```
Comment (josephsavona) [inline src/RelayPublic.js]: same here!

Comment (tlvenn) [inline src/RelayPublic.js]: fixed.

**Comment by josephsavona** [comment]
@facebook-github-bot import

**Pull request closed without merging**